### PR TITLE
Need a logger class to log errors and exceptions #96

### PR DESCRIPTION
**Pull request by lak-proddev**
Signed-off-by: Lakshminarayana Nekkanti <narayana.nekkanti@gmail.com>

### CODE REVIEW

**Comment by jonas-jonas** [comment]
Lars has a little bit of a grudge against Activators. That's why we didn't use them. I'm not sure what the benefit of this over Slf4j/Logback is? Could you explain?

**Comment by vogella** [comment]
Activators should be avoided as they delay the startup of the IDE. We could
use the Logger service from platform. Access (from memory)
Platform.getWorkbench().getService(Logger.class)

Jonas Hungershausen <notifications@github.com> schrieb am Fr., 12. Juli
2019, 10:47:

**Comment by lak-proddev** [comment]
> Lars has a little bit of a grudge against Activators. That's why we didn't use them. I'm not sure what the benefit of this over Slf4j/Logback is? Could you explain?

I supposed to ask about "Activator" before the push. I had a doubt about it not creating yet.. :)
Not really any benefit, I just created a conventional logger used for a plugin.  I have created a convenient method to change logger easily.

**Comment by lak-proddev** [comment]
> Activators should be avoided as they delay the startup of the IDE. We could use the Logger service from platform. Access (from memory) Platform.getWorkbench().getService(Logger.class) Jonas Hungershausen <notifications@github.com> schrieb am Fr., 12. Juli 2019, 10:47:

I will change it.

**Comment by lak-proddev** [comment]
> Activators should be avoided as they delay the startup of the IDE. We could use the Logger service from platform. Access (from memory) Platform.getWorkbench().getService(Logger.class) Jonas 

Lars, You mean? 
`Logger logger = PlatformUI.getWorkbench().getService(org.eclipse.e4.core.services.log.Logger.class);
`
If so, we need to add e4 dependency. Is it fine ?

**Comment by vogella** [comment]
+1

Lakshminarayana Nekkanti <notifications@github.com> schrieb am Fr., 12.
Juli 2019, 12:55:

**Comment by jonas-jonas** [comment]
Yeah, just add the e4 dependency in this patch. I think e4 is an Eclipse project as well, we don't even need to wait for a CQ. 

If I understood correctly, we could then remove the SLF4J and Logback dependencies? If so remove them in this patch as well.

**Comment by lak-proddev** [comment]
The  only concern using `Logger logger = PlatformUI.getWorkbench().getService(org.eclipse.e4.core.services.log.Logger.class);
` is logging with  **org.eclipse.e4.ui.workbench** plugin.I think It should be  **org.eclipse.dartboard**
!ENTRY org.eclipse.e4.ui.workbench 2 0 2019-07-13 15:27:16.840
!MESSAGE test warn

!ENTRY org.eclipse.e4.ui.workbench 4 0 2019-07-13 15:27:44.886
!MESSAGE test error

I tried to fix this. I think the only way is overriding the default context.

**Comment by vogella** [comment]
Maybe add a method to Logger which allows to specify the tag?

Lakshminarayana Nekkanti <notifications@github.com> schrieb am Sa., 13.
Juli 2019, 13:21:

**Comment by lak-proddev** [comment]
> Maybe add a method to Logger which allows to specify the tag? 

Yes, we can add. This is the change required  in **platform.ui** repository.

https://git.eclipse.org/r/#/c/146037/1/bundles/org.eclipse.e4.ui.workbench/src/org/eclipse/e4/ui/internal/workbench/WorkbenchLogger.java
If it's fine for you, I will a file new one in Bugzilla and push the change.

File: org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/*******************************************************************************
```
Comment (jonas-jonas) [inline org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: I don't think we need a seperate Logger class. Can't we just expose the getLogger method and call that in every file? And use org.eclipse.e4.core.services.log.Logger directly?

Comment (lak-proddev) [reply org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: Yes, we can expose the method and call. Calling direct method will show restriction warnings on each file.

Comment (jonas-jonas) [reply org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: Ah, I see. Any way around that, except wrapping it in our own Logger? Like is there a plugin logger? If not. We should keep the Logger class.

Comment (lak-proddev) [reply org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: No Plugin is logger available.  
`BundleContext `will be` null ` for the plugins which don't have the `Activator` class. So, not able to set or get logger for our bundle.

Comment (jonas-jonas) [reply org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: @vogella could we use an Activator for logging purposes? I really dislike the "incouraged access" or the separate Logger class. Seems like the best way would be an Activator.

File: org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java [hunk 2]
```diff
@@ -20,6 +25,18 @@
 	private static org.eclipse.e4.core.services.log.Logger getLogger() {
 		if (logger == null) {
 			logger = PlatformUI.getWorkbench().getService(org.eclipse.e4.core.services.log.Logger.class);
+			Bundle bundle = Platform.getBundle("org.eclipse.e4.ui.workbench"); //$NON-NLS-1$
```
Comment (lak-proddev) [inline org.eclipse.dartboard/src/org/eclipse/dartboard/util/Logger.java]: Using Reflection API. without changing the upstream code.

**Comment by jonas-jonas** [comment]
Replaced with #128

**Pull request closed without merging**